src/jit/simdcodegenxarch.cpp

```diff
@@ -619,8 +619,8 @@ CodeGen::genSIMDIntrinsicInit(GenTreeSIMD* simdNode)
             // data section using movss/sd.  Similarly if op1 is a memory op we
             // would have loaded it using movss/sd.  Movss/sd when loading a xmm reg
             // from memory would zero-out upper bits. In these cases we can
-            // avoid explicitly zero'ing out targetReg.
-            bool zeroInitRequired = !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
+            // avoid explicitly zero'ing out targetReg if targetReg and op1Reg are the same.
+            bool zeroInitRequired = (targetReg != op1Reg) || !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
```

Comment:
Note that by necessiating zeroInit when targetReg != op1Reg we are unnecessarily pessimizing code.

In this case there is no need to use genSIMDScalarMove() because we know the type of value we are moving from op1reg to targetReg is TYP_FLOAT.  The right fix here is

if (!zeroInitRequired)
   ins = ins_Copy(TYP_FLOAT)
else
   ins = ins_Store(TYP_FLOAT)

inst_RV_RV(ins, targetReg, op1Reg, emitTypeSize(baseType));